src/expand.ts

```diff
@@ -150,7 +151,9 @@ async function expandStringHelper(input: string, opts: ExpansionOptions) {
             }
         }
     }
-
+    if (subs.has("${sourceDir}")) {
```

Comment:
Does this need to be checked earlier?  I would think that the warning on line 148 would print out first.